bridge/packages/core/data/lib/src/repository/auth_repository.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:async';
+
+import 'package:core_common/common.dart';
+import 'package:core_datastore/datastore.dart';
+import 'package:core_network/network.dart';
+
+enum AuthenticationStatus { unknown, authenticated, unauthenticated }
+
+class AuthenticationRepository implements Disposable {
+  final _controller = StreamController<AuthenticationStatus>.broadcast();
+
+  final _networkSource = AuthNetworkDataSource();
+  final _tokenLocalSource = TokenLocalDataSource();
+  final _userLocalSource = UserLocalDataSource();
+
+  Stream<AuthenticationStatus> get status async* {
+    await Future<void>.delayed(const Duration(seconds: 1));
+
+    final token = await _tokenLocalSource.getToken();
+    if (token == null) {
+      yield AuthenticationStatus.unauthenticated;
+    } else {
+      yield AuthenticationStatus.authenticated;
+    }
+
+    yield* _controller.stream;
+  }
+
+  Future<void> login({
+    required String email,
+    required String password,
+  }) async {
+    await _networkSource
+        .login(email: email, password: password)
+        .then((result) async {
```

Comment:
I think synchronous syntax would be cleaner here instead of using callbacks.

Reply:
Indeed!